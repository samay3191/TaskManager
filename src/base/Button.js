import React from "react";
import PropTypes from "prop-types";

const Button = ({ text, onClick, disabled }) => {
  return (
    <div className="m-1 w-12">
      <button
        type="button"
        onClick={onClick}
        className={`bg-blue-800 h-full text-xs text-white rounded-lg py-2 px-3 leading-3 align-middle ${disabled ? "opacity-25" : ""}`}
        disabled={disabled}
      >
        {text}
      </button>
    </div>
  );
};

Button.defaultProps = {
  disabled: false
}

Button.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  disabled: PropTypes.bool.isRequired
};

export default Button;
