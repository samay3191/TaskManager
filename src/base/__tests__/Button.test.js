import React from "react";
import { shallow } from "enzyme";
import Button from "../Button";

const initialProps = {
  text: "Submit",
  onClick: jest.fn(),
};

let component;

describe("<Button />", () => {
  beforeAll(() => {
    component = shallow(<Button {...initialProps} />);
  });

  it("should render Button component", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render simulate button click", () => {
    component.find("button").simulate("click");
    expect(initialProps.onClick).toHaveBeenCalled();
  });
});
