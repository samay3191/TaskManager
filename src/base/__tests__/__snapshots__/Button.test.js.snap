// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render Button component 1`] = `
<div
  className="m-1 w-12"
>
  <button
    className="bg-blue-800 h-full text-xs text-white rounded-lg py-2 px-3 leading-3 align-middle "
    disabled={false}
    onClick={[MockFunction]}
    type="button"
  >
    Submit
  </button>
</div>
`;
